napari/utils/dask.py

```diff
@@ -0,0 +1,185 @@
+
+    Returns
+    -------
+    [type]
+        [description]
+    """
+    import psutil
+
+    if nbytes is None:
+        nbytes = psutil.virtual_memory().total * mem_fraction
+    if not (
+        hasattr(utils, 'dask_cache') and isinstance(utils.dask_cache, Cache)
+    ):
+        utils.dask_cache = Cache(nbytes)
+        utils.dask_cache.register()
+    return utils.dask_cache
+
+
+def resize_dask_cache(
+    nbytes: Optional[int] = None, mem_fraction: float = None
+) -> Cache:
+    """Create or resize the dask cache used for opportunistic caching.
+
+    The cache object is an instance of a :class:`Cache`, (which
+    wraps a :class:`cachey.Cache`), and is made available at
+    :attr:`napari.utils.dask_cache`.
+
+    See `Dask opportunistic caching
+    <https://docs.dask.org/en/latest/caching.html>`_
+
+    Parameters
+    ----------
+    nbytes : int, optional
+        The desired size of the cache, in bytes.  If ``None``, the cache size
+        will autodetermined as fraction of the total memory in the system,
+        using ``mem_fraction``.  If ``nbytes`` is 0. The cache is turned off.
+        by default, cache size is autodetermined using ``mem_fraction``.
+    mem_fraction : float, optional
+        The fraction (from 0 to 1) of total memory to use for the dask cache.
+
+    Returns
+    -------
+    dask_cache : Cache
+        An instance of a Dask Cache
+
+    Example
+    -------
+    >>> from napari.utils import resize_dask_cache
+    >>> cache = resize_dask_cache()  # use 50% of total memory by default
+
+    >>> # dask.Cache wraps cachey.Cache
+    >>> assert isinstance(cache.cache, cachey.Cache)
+
+    >>> # useful attributes
+    >>> cache.cache.available_bytes  # full size of cache
+    >>> cache.cache.total_bytes   # currently used bytes
+    """
+
+    import psutil
+
+    if nbytes is None and mem_fraction is not None:
+        nbytes = psutil.virtual_memory().total * mem_fraction
+
+    # if we don't have a cache already, create one.  If neither nbytes nor
+    # mem_fraction was provided, it will use the default size as determined in
+    # create_cache.
+    if not (
+        hasattr(utils, 'dask_cache') and isinstance(utils.dask_cache, Cache)
+    ):
+        return create_dask_cache(nbytes)
+    else:  # we already have a cache
+        # if the cache has already been registered, then calling
+        # resize_dask_cache() without supplying either mem_fraction or nbytes
+        # is a no-op:
+        if (
+            nbytes is not None
+            and nbytes != utils.dask_cache.cache.available_bytes
+        ):
+            utils.dask_cache.cache.resize(nbytes)
+
+    return utils.dask_cache
+
+
+def _is_dask_data(data) -> bool:
+    """Return True if data is a dask array or a list/tuple of dask arrays."""
+    return isinstance(data, da.Array) or (
+        isinstance(data, (list, tuple))
+        and any(isinstance(i, da.Array) for i in data)
+    )
+
+
+def configure_dask(data) -> ContextManager[dict]:
```

Comment:
```suggestion
def configure_dask(data) -> Callable[[], ContextManager[dict]]:
```
mind fixing this for me while you're at it?